registry/client.go

```diff
@@ -949,9 +949,20 @@ func (c *client) IsOperationalScanner(scannerID string) (bool, error) {
 	return contracts.ScannerPoolReg.IsScannerOperational(c.opts, common.HexToAddress(scannerID))
 }
 
-func (c *client) GetActiveScannerStake(scannerID string) (*big.Int, error) {
+func (c *client) GetActiveScannerStake(blockNumber *big.Int, scannerID string) (*big.Int, error) {
+
+	var opts bind.CallOpts
+	if c.opts != nil {
+		opts = *c.opts
+		opts.BlockNumber = blockNumber
+	} else {
+		opts = bind.CallOpts{
+			BlockNumber: blockNumber,
+		}
+	}
```

Comment:
```go
	var opts bind.CallOpts
	if c.opts != nil {
		opts = *c.opts
		opts.BlockNumber = blockNumber
	} else {
		opts = bind.CallOpts{
			BlockNumber: blockNumber,
		}
	}
```
This code block seems to repeat a few times. I guess we could do one of the two things:
- Move this logic to an internal method of `client` like `func (c *client) getBlockOpts(blockNumber *big.Int) *bind.CallOpts`
- Use `PegBlock()` to set `c.opts` and then for each of the methods modified in this PR, we can remove the `blockNumber` argument and use `c.opts` directly

Let's make sure we generate the mocks once more if we do the second.

Reply:
I will proceed to DRY the code